py/gc.c

```diff
@@ -680,6 +692,10 @@ void *gc_alloc(size_t n_bytes, unsigned int alloc_flags) {
         area->gc_last_free_atb_index = (i + 1) / BLOCKS_PER_ATB;
     }
 
+    if (area->gc_high_watermark < end_block) {
```

Comment:
Prefer to write as `area->gc_high_watermark = MAX(area->gc_high_watermark, end_block);` as I feel that conveys the intent better.